arangod/Replication2/StateMachines/Document/MaintenanceActionExecutor.cpp

```diff
@@ -79,7 +80,32 @@ auto MaintenanceActionExecutor::executeDropCollectionAction(
   if (work) {
     return {TRI_ERROR_INTERNAL};
   }
-  return {};
+  return dropCollectionAction.result();
+}
+
+auto MaintenanceActionExecutor::executeModifyCollectionAction(
+    ShardID shard, CollectionID collection,
+    std::shared_ptr<VPackBuilder> properties, std::string followersToDrop)
+    -> Result {
+  namespace maintenance = arangodb::maintenance;
+
+  maintenance::ActionDescription actionDescription(
+      std::map<std::string, std::string>{
+          {maintenance::NAME, maintenance::UPDATE_COLLECTION},
+          {maintenance::DATABASE, _gid.database},
+          {maintenance::COLLECTION, collection},
+          {maintenance::SHARD, shard},
+          {maintenance::SERVER_ID, _server},
+          {maintenance::FOLLOWERS_TO_DROP, followersToDrop}},
+      maintenance::HIGHER_PRIORITY, true, std::move(properties));
+
+  maintenance::UpdateCollection updateCollectionAction(_maintenanceFeature,
+                                                       actionDescription);
+  bool work = updateCollectionAction.first();
+  if (work) {
+    return {TRI_ERROR_INTERNAL};
+  }
```

Comment:
```suggestion
  updateCollectionAction.first();
```
I think you want to ignore the return value, right?

Reply:
I would be curious to know what the return value means. But, since we got the `result()`, I guess we can safely ignore that.

Reply:
Afair this has to do with how the Maintenance handles re-running an action.